forte/pipeline.py

```diff
@@ -401,6 +467,21 @@ def initialize(self) -> "Pipeline":
             self.reader.set_profiling(True)
             self._profiler = [0.0] * len(self.components)
 
+        # Check record types and attributes of each pipeline component
+        if self._do_init_type_check:
+            current_records: Dict[str, Set[str]] = {}
+            self._reader.record(current_records)
+            for component in self.components:
+                if not (
+                    hasattr(component, "expected_types_and_attributes")
+                    and hasattr(component, "record")
```

Comment:
Can we do these two `if` conditions separately? That means, run the two statements for each different condition.